File: src/main/java/io/github/binaryfoo/decoders/SometimesAsciiPrimitiveDecoder.java
package io.github.binaryfoo.decoders;

import io.github.binaryfoo.tlv.ISOUtil;

public class SometimesAsciiPrimitiveDecoder implements PrimitiveDecoder {
    @Override
    public String decode(String hexString) {
        StringBuilder builder = new StringBuilder();
        for (byte b : ISOUtil.hex2byte(hexString)) {
            if (Character.isISOControl(b)) {
                return hexString;
            }
            builder.append((char) b);
        }
        return builder.toString();
    }
}

File: src/main/java/io/github/binaryfoo/decoders/SometimesAsciiPrimitiveDecoder.kt
package io.github.binaryfoo.decoders

import io.github.binaryfoo.tlv.ISOUtil

class SometimesAsciiPrimitiveDecoder : PrimitiveDecoder {
    override fun decode(hexString: String): String {
        val builder = StringBuilder()
        for (b in ISOUtil.hex2byte(hexString)) {
            if (Character.isISOControl(b.toInt())) {
                return hexString
            }
            builder.append(b.toChar())
        }
        return builder.toString()
    }
}
